src/ucx_plugin.c

```diff
@@ -685,152 +710,175 @@ ncclResult_t nccl_ucx_isend(void *send_comm, void *data, int size, int tag, void
     }
   }
 
-#if UCP_API_VERSION >= UCP_VERSION(1, 10)
-  ucp_request_param_t params;
+  req = ucx_request_get(comm);
+  if (req == NULL) {
+    return ncclInternalError;
+  }
+
+  ucx_request_add(req, size);
+
   params.op_attr_mask = UCP_OP_ATTR_FIELD_CALLBACK |
-                        UCP_OP_ATTR_FIELD_DATATYPE;
-  params.datatype     = ucp_dt_make_contig(1);
+                        UCP_OP_ATTR_FIELD_USER_DATA;
   params.cb.send      = send_handler_nbx;
+  params.user_data    = &req->pending;
   if (mh) {
-    params.memory_type  = mh->mem_type;
     params.op_attr_mask |= UCP_OP_ATTR_FIELD_MEMORY_TYPE;
+    params.memory_type   = mh->mem_type;
```

Comment:
if we anyway use "modern" UCP API here why not to pass memory handle for preregistered buffers?

Reply:
let's postpone this change, as it may imply additional limitations

Reply:
plan is to check that independently from multi-recv fix. but for my understanding, currently, the memh is found internally through rcache from passed addr/len, is that correct?